corgi/collectors/models.py

```diff
@@ -88,3 +88,54 @@ class CollectorRPMRepository(models.Model):
 
     def __str__(self):
         return f"{self.name}"
+
+
+class CollectorPyxisModel(models.Model):
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    meta_attr = models.JSONField(default=dict)
+
+    class Meta:
+        abstract = True
+        ordering = ["uuid"]
+
+
+class CollectorPyxisImage(CollectorPyxisModel):
+    # This corresponds to the sha256 hash we use to lookup the image index or "noarch" container eg:
+    # pkg:oci/ose-ovirt-csi-driver-operator@
+    # --> sha256:2b55f17547ddcbdbefddde7372bc3ddfba9b66dfe76d453e479f9efd51514656 <--
+    # ?repository_url=registry.redhat.io/openshift4/ose-ovirt-csi-driver-operator
+    # &tag=v4.12.0-202301042354.p0.gfeb14fb.assembly.stream
+    manifest_list_digest = models.CharField(max_length=200)
+    # Pyxis doesn't have a distinct Image for the "noarch" container like Corgi has, it only has
+    # image objects for the arch specific containers, so we mirror that here. This value represents
+    # the sha256 hash from the arch specific container purl version eg:
+    # pkg:oci/ose-ovirt-csi-driver-operator@
+    # --> sha256:596d18e9a4be00f7c5f1ef536b0d34b3a3a5b826807650cc866bbb7976b237b4 <--
+    # ?arch=aarch64
+    # &repository_url=registry.redhat.io/openshift4/ose-ovirt-csi-driver-operator
+    # &tag=v4.12.0-202301042354.p0.gfeb14fb.assembly.stream
+    image_id = models.CharField(max_length=200)
+    arch = models.CharField(max_length=20, default="")
+    nvr = models.TextField()
+    name_label = models.CharField(max_length=200, default="")
+    tags = fields.ArrayField(models.CharField(max_length=200), default=list)
+    creation_date = models.DateTimeField(null=True)
+    pyxis_id = models.CharField(unique=True, max_length=50)
+
+
+class CollectorPyxisImageRepository(CollectorPyxisModel):
+    name = models.CharField(max_length=200)
+    registry = models.CharField(max_length=200)
+    images = models.ManyToManyField(CollectorPyxisImage, related_name="repos")
+    image_advisory_id = models.CharField(max_length=50, default="")
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_pyxis_repo",
+                fields=["registry", "name"],
+            )
+        ]
+
+    def __str__(self):
+        return f"{self.name}"
```

Comment:
If we expect the combo of registry and name to be unique together, maybe it's better to return that in our string representation instead of just the name.
```suggestion
    def __str__(self):
        return f"{self.registry}/{self.name}"
```

Reply:
done